assets/queries/cloudFormation/iam_user_passwordResetRequired_false/query.rego

```diff
@@ -0,0 +1,44 @@
+package Cx
+
+CxPolicy [result] {
+  resource := input.document[i].Resources[name]
+  resource.Type == "AWS::IAM::User"
+  loginProfile := resource.Properties.LoginProfile
+  loginProfile != []
+  loginProfile != null
```

Comment:
Isn't LoginProfile an object? Since in this test case you are assuming its present (or not testing its presence), you won't need to check if it is defined

Reply:
Hi João.
I've done this for the follow example
Properties:
       Path: "/"
       LoginProfile:
         Password: myP@ssW0rd
         PasswordResetRequired: true
       Policies: ...
         
This is what we are expecting. However, a user can define this object empty like this:
Properties:
        Path: "/"
        LoginProfile:
        Policies:
        ...


It's a valid case where a password it's not defined for a user for example at this stage but there is a password defined for the group where this user is inserted in.
So I've not only checked for the object presence but also if it has content.

Reply:
I'm still a bit skeptical about having LoginProfile empty, if there's nothing to add why place it? The Group password won't be referenced by some user id, not having anything to do with this LoginProfile? Could you please provide a link or reference to that example you mention so I can dig a little deeper on this issue?

Reply:
Hi João :)

after our discussion in Teams, I've changed this query since I did not checked the next line after these 2.
This solution was not well implemented :)
I've already done new changes